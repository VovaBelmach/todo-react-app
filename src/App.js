import Todos from "./components/Todos/Todos";

import styles from "./App.module.css";

function App() {
  return (
    <>
      <div className={`${styles["app-header"]}`} />
      <div className={styles.app}>
        <main className={`${styles["app-content"]}`}>
          <header>
            <h1 className={styles.title}>TODO</h1>
          </header>
          <Todos />
          <footer>
            <p className={styles.footer}>Drag and drop to reorder list</p>
          </footer>
        </main>
      </div>
    </>
  );
}

export default App;