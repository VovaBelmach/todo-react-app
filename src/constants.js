// LocalStorage Todo name
export const TODOS_LOCAL_STORAGE_NAME = "todos";

// TodoFilter component
export const TODO_FILTER_ALL_BUTTON_NAME = "All";
export const TODO_FILTER_ACTIVE_BUTTON_NAME = "Active";
export const TODO_FILTER_COMPLITED_BUTTON_NAME = "Completed";

// TodoFooter component
export const TODO_FOOTER_ALL_DONE_TEXT = "All done!";
export const TODO_FOOTER_NUMBER_TODO_LEFT_TEXT = "todo left";
export const TODO_FOOTER_CLEAR_COMPLETED_BUTTON_NAME = "Clear completed";

// TodoInput component
export const TODO_INPUT_EMPTY_ERROR_MESSAGE = "Oops! You cannot create an empty todo. Please provide 'What should be done?' in the section above.";
export const TODO_INPUT_PLACEHOLDER_TEXT = "What should be done?";

// TodoItem component
export const TODO_ITEM_DELETE_BUTTON_NAME = "X";

// TodoList component
export const TODO_LIST_EMPTY_LIST_MESSAGE = "All todos completed! Add a todo and it will be displayed here.";

// App component
export const HEADER_TEXT = "TODO";
export const FOOTER_TEXT = "Drag and drop to reorder list.";