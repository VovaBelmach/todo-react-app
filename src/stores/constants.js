export const TODOS_LOCAL_STORAGE_NAME = "todos";