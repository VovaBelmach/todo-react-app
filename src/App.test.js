import { render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import App from "./App";
import store from "./stores/configureStore";
import { HEADER_TEXT, FOOTER_TEXT } from "./constants";

describe("App", () => {
  it("should render the header and footer text", () => {
    render(
      <Provider store={store}>
        <App />
      </Provider>
    );
    expect(screen.getByText(HEADER_TEXT)).toBeInTheDocument();
    expect(screen.getByText(FOOTER_TEXT)).toBeInTheDocument();
  });

  it("should render the correct CSS classes", () => {
    render(
      <Provider store={store}>
        <App />
      </Provider>
    );
    const appHeader = screen.getByTestId("app-header");
    const app = screen.getByTestId("app");
    const appContent = screen.getByTestId("app-content");
    const title = screen.getByTestId("title");
    const footer = screen.getByTestId("footer");

    expect(appHeader).toHaveClass("app-header");
    expect(app).toHaveClass("app");
    expect(appContent).toHaveClass("app-content");
    expect(title).toHaveClass("title");
    expect(footer).toHaveClass("footer");
  });
});
