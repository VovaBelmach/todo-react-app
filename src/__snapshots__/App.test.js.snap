// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should renders correctly 1`] = `
Array [
  <div
    className="app-header"
    data-testid="app-header"
  />,
  <div
    className="app"
    data-testid="app"
  >
    <main
      className="app-content"
      data-testid="app-content"
    >
      <header>
        <h1
          className="title"
          data-testid="title"
        >
          TODO
        </h1>
      </header>
      <div
        className="card todo-input"
      >
        <form
          aria-label="form"
          onSubmit={[Function]}
        >
          <label>
            <input
              className="checkbox"
              disabled={true}
              type="checkbox"
            />
          </label>
          <input
            name="todo-input"
            onChange={[Function]}
            placeholder="What should be done?"
            type="text"
            value=""
          />
        </form>
        <span
          className="error-message"
        />
      </div>
      <div
        className="card todo-list"
      >
        <p>
          All todos completed! Add a todo and it will be displayed here.
        </p>
        <footer
          className="todo-footer"
        >
          <span>
            All done!
          </span>
          <ul
            className="todo-filter"
          >
            <li>
              <button
                className="button "
                onClick={[Function]}
                type="button"
                value="All"
              >
                All
              </button>
            </li>
            <li>
              <button
                className="button "
                onClick={[Function]}
                type="button"
                value="Active"
              >
                Active
              </button>
            </li>
            <li>
              <button
                className="button "
                onClick={[Function]}
                type="button"
                value="Completed"
              >
                Completed
              </button>
            </li>
          </ul>
          <button
            className="button todo-clear-completed"
            onClick={[Function]}
            type="button"
            value="Clear completed"
          >
            Clear completed
          </button>
        </footer>
      </div>
      <footer>
        <p
          className="footer"
          data-testid="footer"
        >
          Drag and drop to reorder list.
        </p>
      </footer>
    </main>
  </div>,
]
`;
